src/cursor/cur_std.c

```diff
@@ -1175,13 +1179,76 @@ __wt_cursor_bound(WT_CURSOR *cursor, const char *config)
+                WT_ERR(__wt_compare(
+                  session, CUR2BT(cursor)->collator, &key, &cursor->lower_bound, &exact));
+                if (exact < 0)
+                    WT_ERR_MSG(session, EINVAL, "The provided cursor bounds are overlapping");
+            }
+            /* Copy the key over to the upper bound item and set upper bound and inclusive flags. */
+            F_SET(cursor, WT_CURSTD_BOUND_UPPER);
+            WT_ERR(__wt_config_gets(session, cfg, "inclusive", &cval));
+            if (cval.val)
+                F_SET(cursor, WT_CURSTD_BOUND_UPPER_INCLUSIVE);
+            WT_ERR(__wt_buf_set(session, &cursor->upper_bound, key.data, key.size));
+        } else if (WT_STRING_MATCH("lower", cval.str, cval.len)) {
+            /*
+             * If the upper bounds are set, make sure that the lower bound is less than the upper
+             * bound.
+             */
+            WT_ERR(__wt_cursor_get_raw_key(cursor, &key));
+            if (F_ISSET(cursor, WT_CURSTD_BOUND_UPPER)) {
+                WT_ERR(__wt_compare(
+                  session, CUR2BT(cursor)->collator, &key, &cursor->upper_bound, &exact));
+                if (exact > 0)
+                    WT_ERR_MSG(session, EINVAL, "The provided cursor bounds are overlapping");
+            }
+            /* Copy the key over to the lower bound item and set upper bound and inclusive flags. */
+            F_SET(cursor, WT_CURSTD_BOUND_LOWER);
+            WT_ERR(__wt_config_gets(session, cfg, "inclusive", &cval));
+            if (cval.val)
+                F_SET(cursor, WT_CURSTD_BOUND_LOWER_INCLUSIVE);
+            WT_ERR(__wt_buf_set(session, &cursor->lower_bound, key.data, key.size));
+        } else
+            WT_ERR_MSG(session, EINVAL,
+              "setting bounds only accepts \"upper\" or \"lower\" as the configuration");
+
     } else if (WT_STRING_MATCH("clear", cval.str, cval.len)) {
-        /* Inclusive should not be supplied from the application the action clear configuration. */
+        /* Inclusive should not be supplied from the application with action clear configuration. */
         if (__wt_config_getones(session, config, "inclusive", &cval) != WT_NOTFOUND)
             WT_ERR_MSG(session, EINVAL,
               "clearing bounds is not compatible with the inclusive configuration");
-    }
 
+        /*
+         * Check if there is a lower or upper specified bound config. If there are no specified
+         * bounds, both the upper and lower bound will be cleared.
+         */
+        WT_ERR(__wt_config_gets(session, cfg, "bound", &cval));
+        if (cval.len == 0) {
+            F_CLR(cursor, WT_CURSTD_BOUNDS_ALL);
+            __wt_buf_free(session, &cursor->upper_bound);
+            WT_CLEAR(cursor->upper_bound);
+            __wt_buf_free(session, &cursor->lower_bound);
+            WT_CLEAR(cursor->lower_bound);
+        } else if (WT_STRING_MATCH("upper", cval.str, cval.len)) {
```

Comment:
There's a way to remove repeated code here.  Remove the first clause (`if (cval.len == 0)`) entirely.  Then:
```
  if (cval.len == 0 || WT_STRING_MATCH("upper", cval.str, cval.len)) {
     ...
  }
  if (cval.len == 0 || WT_STRING_MATCH("lower", cval.str, cval.len)) {
     ...
  }
```
(No else clause).  Feel free to ignore if you don't like that approach.